### ISSUE TITLE
Infinite height resize on iOS Webkit 

### ISSUE THREAD

**Issue by mwgdavies** [google/blockly#92]
Hi

I am using resizable blockly for a cross device visual IDE. Things work great on most devices but I'm having problems with Chrome and Safari on iOS.

On load, blockly appears as expected but as soon as I open a toolbox menu the ` blocklySvg ` element and several of its children go into an infinite height resize. I noticed that https://developers.google.com/blockly/ seems to have the same problem.

I assume this is an event loop issue somewhere deep in one of the many Google dependencies but I can't get to the bottom of it :(

**Comment by NeilFraser** [comment]
Oh great.  That bug is back.  iOS is like a game of Whack-a-Mole.
Thanks, I'll see what's up this time.

**Comment by NeilFraser** [comment]
As an immediate fix, you can delete the +1 on this line:
  treeDiv.style.height = (svgSize.height + 1) + 'px';
in core/toolbox.js.

However, that +1 is there for a reason, and its removal creates a one-pixel gap in other circumstances.  Working on a proper fix...

**Comment by NeilFraser** [comment]
Fixed, I hope for good this time:
https://github.com/google/blockly/commit/29976c4c4d81ed72b4152aa918dee2afd199f2fe

The root issue is that an image (Blockly is an SVG image) that is 100x100 and has a 1 pixel border is actually 102x102.  This causes a Blockly instance that is sized to be 100% of its container's size to actually size itself to 100% plus two pixels.  That causes the container to increase its size by two pixels to hold Blockly.  Whereupon Blockly increases by another two pixels.  Rinse.  Repeat.

Most browsers are smart enough to not get into these infinite growth loops.  But iOS Safari is ... special.  I've removed the external borders from Blockly, and am now painting them inside Blockly's image.  The issue should be resolved.

Thanks for the report!
